packages/react-components/react-dialog/src/components/DialogTitle/useDialogTitle.tsx

```diff
@@ -17,21 +18,21 @@ import { useDialogTitleInternalStyles } from './useDialogTitleStyles.styles';
  * @param ref - reference to root HTMLElement of DialogTitle
  */
 export const useDialogTitle_unstable = (props: DialogTitleProps, ref: React.Ref<HTMLElement>): DialogTitleState => {
-  const { as, action } = props;
+  const { as = defaultComponentType, action } = props;
   const modalType = useDialogContext_unstable(ctx => ctx.modalType);
   const internalStyles = useDialogTitleInternalStyles();
 
   return {
-    components: {
-      root: 'h2',
-      action: 'div',
-    },
-    root: getNativeElementProps(as ?? 'h2', {
-      ref,
-      id: useDialogContext_unstable(ctx => ctx.dialogTitleId),
-      ...props,
-    }),
-    action: resolveShorthand(action, {
+    root: slot<DialogTitleProps>(
+      getNativeElementProps(as, {
+        ref,
+        id: useDialogContext_unstable(ctx => ctx.dialogTitleId),
+        ...props,
+      }),
+      { componentType: defaultComponentType, required: true },
```

Comment:
```ts
      getNativeElementProps(as, {}),
      { componentType: defaultComponentType, required: true },
```

This smells a bit, looks we can consider to create a special function for `root` slot or embed props filtering to `slot()`

Reply:
Or we could just stop prop filtering?! 👀 as we have concise types to avoid unnecessary properties?!

Reply:
It makes absolutely no sense to embed props filtering to `slot`, it has proven to be unnecessary for every single slot and that is precisely why we don't have it in all implementations, I do believe we can move forward to a way that we stop treating `root` as something different than just another slot, let's just stop prop filtering root.

Reply:
> What are the impediments stopping us from opting out of not filtering props for the root slot?

We will need to filter out component's props from `props` manually, which might not be so bad.